cmake/SpectreAddInterfaceLibraryHeaders.cmake

```diff
@@ -52,6 +52,34 @@ function(add_interface_lib_headers)
   # elements of a list.
   string(REPLACE ";" ":" TARGET_HEADERS
     "${ARG_TARGET}=${ARG_HEADERS}")
+
+  # Get already added headers and
```

Comment:
Could you expand this commend a bit more? This is quite the list of operations being done

Reply:
Hah, yea, agreed. I just pushed an update (squashed) that attempts to explain what this is doing.